test/TextUpdater.swift

```diff
@@ -1,27 +1,48 @@
 import UIKit
 
-public enum TextUpdater {
+protocol ViewModel {
+    typealias Services
+    typealias Presenters
     
-    struct ViewModelPresenter {
+    static func ViewModel(services: Services, _ presenters: Presenters);
+}
+
+extension ViewModel {
+    typealias UnpresentedViewModel = Presenters -> ()
+    static func ViewModel(services: Services)(presenters: Presenters) {
+        ViewModel(services, presenters)
+    }
+}
+
+enum TextUpdater : ViewModel {
+    struct Services {
+        let reversText: String -> String
+    }
+    
+    struct Presenters {
         let text: Presenter<String>
-        let input: Presenter<String -> ()>
+        let textUpdate: Presenter<String -> ()>
     }
     
-    @objc(TextUpdaterViewController)
-    public class ViewController: UIViewController {
-        
-        @IBOutlet weak var label: UILabel!
-        @IBOutlet weak var textField: UITextField!
+    static func ViewModel(services: Services, _ presenters: Presenters) {
+        presenters.text.present("")
         
-        var viewModel: ((() -> ViewModelPresenter) -> ())?
-        
-        override public func viewDidLoad() {
-            super.viewDidLoad()
-            
-            viewModel! {[unowned self] in
-                return ViewModelPresenter(
-                    text: self.label.textPresenter,
-                    input:  self.textField.textSinkPresenter) }
+        presenters.textUpdate.present { newText in
+            presenters.text.present(services.reversText(newText))
         }
     }
+}
+
+class TextUpdaterViewController: UIViewController {
+    @IBOutlet weak var label: UILabel!
+    @IBOutlet weak var textField: UITextField!
+    
+    var viewModel: TextUpdater.UnpresentedViewModel?
+    
+    override func viewDidLoad() {
+        viewModel?(TextUpdater.Presenters(
```

Comment:
I meant why not implicitly unwrapped call
so you will know if you forgot to set presenters